<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GalgoR user Guide • galgoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GalgoR user Guide">
<meta property="og:description" content="galgoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">galgoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/galgoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/galgo_callbacks.html">Galgo Callbacks Mechanism</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/harpomaxx/galgoR-package/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>GalgoR user Guide</h1>
                        <h4 class="author">Martin E. Guerrero-Gimenez</h4>
            <address class="author_afil">
      Laboratory of Oncology, Institute of Medicine and Experimental Biology of Cuyo (IMBECU), National Scientific and Technical Research Council (CONICET), Mendoza, Argentina.<br><a class="author_email" href="mailto:#"></a><a href="mailto:mguerrero@mendoza-conicet.gob.ar" class="email">mguerrero@mendoza-conicet.gob.ar</a>
      </address>
                              <h4 class="author">Juan Manuel Fernandez-Muñoz</h4>
            <address class="author_afil">
      Laboratory of Oncology, Institute of Medicine and Experimental Biology of Cuyo (IMBECU), National Scientific and Technical Research Council (CONICET), Mendoza, Argentina.<br><h4 class="author">Carlos A. Catania</h4>
            <address class="author_afil">
      LABSIN, Engineering School, National University of Cuyo, Mendoza, Argentina.<br><a class="author_email" href="mailto:#"></a><a href="mailto:harpo@ingenieria.uncuyo.edu.ar" class="email">harpo@ingenieria.uncuyo.edu.ar</a>
      </address>
                  
            <h4 class="date">2020-07-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/harpomaxx/galgoR-package/blob/master/vignettes/galgoR.rmd"><code>vignettes/galgoR.rmd</code></a></small>
      <div class="hidden name"><code>galgoR.rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>We report a novel method to identify specific transcriptomic phenotypes based on an elitist non-dominated sorting genetic algorithm that combines the advantages of clustering methods and the exploratory properties of genetic algorithms to discover biologically and clinically relevant molecular subtypes in different cancers.</p>
    </div>
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>In the new era of omics data, precision medicine has become the new paradigm of cancer treatment. Among all available omics techniques, gene expression profiling, in particular, has been increasingly used to classify tumor subtypes with different biological behavior. Cancer subtype discovery is usually approached from two possible perspectives:</p>
<ul>
<li>Using the molecular data alone with unsupervised techniques such as clustering analysis.</li>
<li>Using supervised techniques focusing entirely on survival data.</li>
</ul>
<p>The problem of finding patients subgroups with survival differences while maintaining cluster consistency could be viewed as a bi-objective problem, where there is a trade-off between the separability of the different groups and the ability of a given signature to consistently distinguish patients with different clinical outcomes. This gives rise to a set of optimal solutions, also known as Pareto-optimal solutions. To overcome these issues, we combined the advantages of clustering methods for grouping heterogeneous omics data and the search properties of genetic algorithms in galgoR: A flexible yet robust multi-objective meta-heuristic for disease subtype discovery based on an elitist non-dominated sorting genetic algorithm (NSGA-II), driven by the underlying premise of maximizing survival differences between groups while getting high consistency and robustness of the clusters obtained.</p>
</div>
<div id="algorithm" class="section level1">
<h1 class="hasAnchor">
<a href="#algorithm" class="anchor"></a>Algorithm</h1>
<p>In the galgoR package, the NSGA-II framework was used for finding multiple Pareto-optimal solutions to classify patients according to their gene expression patterns. Basically, NSGA-II starts with a population of competing individuals which are evaluated under a set of fitness functions that estimate the survival differences and cohesiveness of the different transcriptomic groups. Then, solutions are ranked and sorted according to their non-domination level which will affect the way they are chosen to be submitted to the so-called “evolutionary operators” such as crossover and mutation. Once a set of well-suited solutions are selected and reproduced, a new offspring of individuals composed of a mixture of the “genetic information” of the parents is obtained. Parents and offspring are pooled and the best-ranked solutions are selected and passed to the next generation which will start over the same process again.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div id="galgor-library" class="section level2">
<h2 class="hasAnchor">
<a href="#galgor-library" class="anchor"></a>galgoR library</h2>
<p>To install galgoR package, start R and enter:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"https://github.com/harpomaxx/galgoR-package"</span>)

<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">galgoR</span>)</pre></body></html></div>
</div>
<div id="examples-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#examples-datasets" class="anchor"></a>Examples datasets</h2>
<p>To standardize the structure of genomic data, we use the <a href="https://www.bioconductor.org/packages/release/bioc/html/Biobase.html">ExpressionSet</a> structure for the examples given in this guide. The <code>ExpressionSet</code> objects are formed mainly by:</p>
<ul>
<li>A matrix of genetic expression, usually derived from microarray or RNAseq experiments.</li>
<li>Phenotypic data, where we find information on the samples (condition, status, treatment, survival, and other covariates).</li>
<li>Finally, these objects can also contain Annotations and feature Meta-data.</li>
</ul>
<p>To start testing galgoR, we will use two Breast Cancer datasets. Namely, the <a href="bioconductor.org/packages/release/data/experiment/html/breastCancerUPP.html">UPP</a> and the <a href="bioconductor.org/packages/release/data/experiment/html/breastCancerTRANSBIG.html">TRANSBIG</a> datasets. Additionally, we will use PAM50 centroids to perform breas cancer sample classification. The datasets can be accesed from the following <a href="https://bioconductor.org/">Bioconductor</a> packages:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r">
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)

<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"breastCancerUPP"</span>,<span class="kw">version</span> <span class="kw">=</span> <span class="st">"devel"</span>)
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"breastCancerTRANSBIG"</span>,<span class="kw">version</span> <span class="kw">=</span> <span class="st">"devel"</span>)</pre></body></html></div>
<div class="sourceCode" id="cb3"><html><body><pre class="r">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">breastCancerTRANSBIG</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">breastCancerUPP</span>)</pre></body></html></div>
<p>Also, some basic packages are needed to run the example in this vignette</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">galgoR</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Biobase</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">genefu</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">survival</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">survminer</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)</pre></body></html></div>
</div>
</div>
<div id="examples" class="section level1">
<h1 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h1>
<div id="loading-data" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-data" class="anchor"></a>Loading data</h2>
<p>To access the <code>ExpressionSets</code> we use:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r">  <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">upp</span>)
  <span class="no">Train</span><span class="kw">&lt;-</span> <span class="no">upp</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="no">upp</span>)

  <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">transbig</span>)
  <span class="no">Test</span><span class="kw">&lt;-</span> <span class="no">transbig</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="no">transbig</span>)

  <span class="co">#To access gene expression data</span>
  <span class="no">train_expr</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Train</span>)
  <span class="no">test_expr</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Test</span>)

  <span class="co">#To access feature data</span>
  <span class="no">train_features</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">Train</span>)
  <span class="no">test_features</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">Test</span>)

  <span class="co">#To access clinical data</span>
  <span class="no">train_clinic</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Train</span>)
  <span class="no">test_clinic</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Test</span>)</pre></body></html></div>
</div>
<div id="data-tidying-and-preparation" class="section level2">
<h2 class="hasAnchor">
<a href="#data-tidying-and-preparation" class="anchor"></a>Data tidying and preparation</h2>
<p>Galgo can accept any numeric data, like probe intensity from microarray experiments or RNAseq normalized counts, nevertheless, features are expected to be scaled across the dataset before being plugged in into the Galgo Framework. For PAM50 classification, Gene Symbols are expected, so probesets are mapped into their respective gene symbols. Probesets mapping for multiple genes are expanded while Genes mapped to multiple probes are collapsed selecting the probes with the highest variance for each duplicated gene.</p>
<div id="drop-duplicates-and-nas" class="section level3">
<h3 class="hasAnchor">
<a href="#drop-duplicates-and-nas" class="anchor"></a>Drop duplicates and NA’s</h3>
<div class="sourceCode" id="cb6"><html><body><pre class="r">
<span class="co">#Custom function to drop duplicated genes (keep genes with highest variance)</span>

<span class="no">DropDuplicates</span><span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">eset</span>, <span class="no">map</span><span class="kw">=</span> <span class="st">"Gene.symbol"</span>){

  <span class="co">#Drop NA's</span>
  <span class="no">drop</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[,<span class="no">map</span>]))
  <span class="no">eset</span> <span class="kw">&lt;-</span> <span class="no">eset</span>[-<span class="no">drop</span>,]

  <span class="co">#Drop duplicates</span>
  <span class="no">drop</span> <span class="kw">&lt;-</span> <span class="kw">NULL</span>
  <span class="no">Dup</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[,<span class="no">map</span>])),<span class="no">map</span>]))
  <span class="no">Var</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">eset</span>),<span class="fl">1</span>,<span class="no">var</span>)
  <span class="kw">for</span>(<span class="no">j</span> <span class="kw">in</span> <span class="no">Dup</span>){
   <span class="no">pos</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[,<span class="no">map</span>]<span class="kw">==</span><span class="no">j</span>)
    <span class="no">drop</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">drop</span>,<span class="no">pos</span>[-<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="no">Var</span>[<span class="no">pos</span>])])

  }
  <span class="no">eset</span> <span class="kw">&lt;-</span> <span class="no">eset</span>[-<span class="no">drop</span>,]

<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">eset</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[,<span class="no">map</span>]
<span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">eset</span>)
}</pre></body></html></div>
<p>###Expand probesets that map for multiple genes</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r">
<span class="co"># Custom function to expand probesets mapping to multiple genes</span>
<span class="no">expandProbesets</span> <span class="kw">&lt;-</span> <span class="kw">function</span> (<span class="no">eset</span>, <span class="no">sep</span> <span class="kw">=</span> <span class="st">"///"</span>, <span class="no">map</span><span class="kw">=</span><span class="st">"Gene.symbol"</span>)
{
  <span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">eset</span>), <span class="kw">function</span>(<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="no">x</span>, <span class="no">sep</span>)<span class="kw">[[</span><span class="fl">1</span>]])
  <span class="no">y</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[,<span class="no">map</span>]), <span class="kw">function</span>(<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="no">x</span>, <span class="no">sep</span>))
  <span class="no">eset</span> <span class="kw">&lt;-</span> <span class="no">eset</span>[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="no">x</span>, <span class="no">length</span>)), ]
  <span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">eset</span>), <span class="kw">function</span>(<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="no">x</span>, <span class="no">sep</span>)<span class="kw">[[</span><span class="fl">1</span>]])
  <span class="no">y</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[,<span class="no">map</span>]), <span class="kw">function</span>(<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="no">x</span>, <span class="no">sep</span>))
  <span class="no">idx</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">x</span>), <span class="kw">function</span>(<span class="no">i</span>) <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="no">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">x</span><span class="kw">[[</span><span class="no">i</span>]]))))
  <span class="no">idy</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">y</span>), <span class="kw">function</span>(<span class="no">i</span>) <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="no">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">y</span><span class="kw">[[</span><span class="no">i</span>]]))))
  <span class="no">xx</span> <span class="kw">&lt;-</span> !<span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">x</span>))
  <span class="no">idx</span> <span class="kw">&lt;-</span> <span class="no">idx</span>[<span class="no">xx</span>]
  <span class="no">idy</span> <span class="kw">&lt;-</span> <span class="no">idy</span>[<span class="no">xx</span>]
  <span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">x</span>)[<span class="no">xx</span>]
  <span class="no">y</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">y</span>)[<span class="no">xx</span>]

  <span class="no">eset</span> <span class="kw">&lt;-</span> <span class="no">eset</span>[<span class="no">idx</span>, ]
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">eset</span>) <span class="kw">&lt;-</span> <span class="no">x</span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)[,<span class="no">map</span>] <span class="kw">&lt;-</span> <span class="no">x</span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>)$<span class="no">gene</span> <span class="kw">&lt;-</span> <span class="no">y</span>
  <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">eset</span>)

}</pre></body></html></div>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">Train</span><span class="kw">=</span><span class="fu">DropDuplicates</span>(<span class="no">Train</span>)
<span class="no">Train</span><span class="kw">=</span><span class="fu">expandProbesets</span>(<span class="no">Train</span>)

<span class="no">Train</span> <span class="kw">&lt;-</span> <span class="no">Train</span>[,!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw pkg">survival</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Train</span>)$<span class="no">t.rfs</span>,<span class="kw">event</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Train</span>)$<span class="no">e.rfs</span>))] <span class="co">#Drop NAs in survival</span>



<span class="no">Test</span><span class="kw">=</span><span class="fu">DropDuplicates</span>(<span class="no">Test</span>)
<span class="no">Test</span><span class="kw">=</span><span class="fu">expandProbesets</span>(<span class="no">Test</span>)
<span class="no">Test</span> <span class="kw">&lt;-</span> <span class="no">Test</span>[,!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw pkg">survival</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Test</span>)$<span class="no">t.rfs</span>,<span class="kw">event</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Test</span>)$<span class="no">e.rfs</span>))] <span class="co">#Drop NAs in survival</span>


<span class="co">#Determine common probes (Genes)</span>
<span class="no">Int</span><span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">Train</span>),<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">Test</span>))

<span class="no">Train</span><span class="kw">=</span> <span class="no">Train</span>[<span class="no">Int</span>,]
<span class="no">Test</span><span class="kw">=</span> <span class="no">Test</span>[<span class="no">Int</span>,]

<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">Train</span>),<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">Test</span>))
<span class="co">#&gt; [1] TRUE</span></pre></body></html></div>
<p>For simplicity and speed, we will create a reduced expression matrix for the examples.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r">
<span class="co">#First we will get PAM50 centroids from genefu package</span>

<span class="no">PAM50Centroids</span> <span class="kw">&lt;-</span> <span class="no">pam50</span>$<span class="no">centroids</span>
<span class="no">PAM50Genes</span> <span class="kw">&lt;-</span> <span class="no">pam50</span>$<span class="no">centroids.map</span>$<span class="no">probe</span>
<span class="no">PAM50Genes</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">Train</span>)[ <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">Train</span>) <span class="kw">%in%</span> <span class="no">PAM50Genes</span>]

<span class="co">#Now we sample 200 random genes from expression matrix</span>

<span class="no">Non_PAM50Genes</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">Train</span>)[ !<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">featureNames</a></span>(<span class="no">Train</span>) <span class="kw">%in%</span> <span class="no">PAM50Genes</span>]
<span class="no">Non_PAM50Genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="no">Non_PAM50Genes</span>,<span class="fl">200</span>, <span class="kw">replace</span><span class="kw">=</span><span class="fl">FALSE</span>)

<span class="no">reduced_set</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">PAM50Genes</span>, <span class="no">Non_PAM50Genes</span>)

<span class="co">#Now we get the reduced training and test sets</span>

<span class="no">Train</span><span class="kw">&lt;-</span> <span class="no">Train</span>[<span class="no">reduced_set</span>,]
<span class="no">Test</span><span class="kw">&lt;-</span> <span class="no">Test</span>[<span class="no">reduced_set</span>,]</pre></body></html></div>
</div>
<div id="rescale-expression-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#rescale-expression-matrix" class="anchor"></a>Rescale expression matrix</h3>
<p>Apply robust linear scaling as proposed in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3283537/#bib61">paper reference</a></p>
<div class="sourceCode" id="cb10"><html><body><pre class="r">
<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Train</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Train</span>),<span class="fl">1</span>,<span class="kw pkg">genefu</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/genefu/man/rescale.html">rescale</a></span>,<span class="kw">na.rm</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">q</span><span class="kw">=</span><span class="fl">0.05</span>))
<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Test</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Test</span>),<span class="fl">1</span>,<span class="kw pkg">genefu</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/genefu/man/rescale.html">rescale</a></span>,<span class="kw">na.rm</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">q</span><span class="kw">=</span><span class="fl">0.05</span>))

<span class="no">train_expr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Train</span>)
<span class="no">test_expr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Test</span>)</pre></body></html></div>
</div>
<div id="survival-object" class="section level3">
<h3 class="hasAnchor">
<a href="#survival-object" class="anchor"></a>Survival Object</h3>
<p>The ‘Surv’ object is created by the Surv() function of the survival package. This uses phenotypic data that are contained in the corresponding datasets, accessed by the <code>pData</code> command.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">train_clinic</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Train</span>)
<span class="no">test_clinic</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Test</span>)

<span class="no">train_surv</span> <span class="kw">&lt;-</span> <span class="kw pkg">survival</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">train_clinic</span>$<span class="no">t.rfs</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">train_clinic</span>$<span class="no">e.rfs</span>)
<span class="no">test_surv</span> <span class="kw">&lt;-</span> <span class="kw pkg">survival</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">test_clinic</span>$<span class="no">t.rfs</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">test_clinic</span>$<span class="no">e.rfs</span>)</pre></body></html></div>
</div>
</div>
<div id="run-galgo" class="section level2">
<h2 class="hasAnchor">
<a href="#run-galgo" class="anchor"></a>Run galgo()</h2>
<p>The main function in this package is <code><a href="../reference/galgo.html">galgo()</a></code>. It accepts an expression matrix and survival object to find robust gene expression signatures related to a given outcome. This function contains some parameters that can be modified, according to the characteristics of the analysis to be performed.</p>
<div id="setting-parameters" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-parameters" class="anchor"></a>Setting parameters</h3>
<p>The principal parameters are:</p>
<ul>
<li>population: a number indicating the number of solutions in the population of solutions that will be evolved</li>
<li>generations: a number indicating the number of iterations of the galgo algorithm</li>
<li>nCV: number of cross-validation sets</li>
<li>usegpu: logical default to FALSE, set to TRUE if you wish to use gpu computing (gpuR package must be properly installed and loaded)</li>
<li>distancetype: character, it can be ‘pearson’ (centered pearson), ‘uncentered’ (uncentered pearson), ‘spearman’ or ‘euclidean’</li>
<li>TournamentSize: a number indicating the size of the tournaments for the selection procedure</li>
<li>period: a number indicating the outcome period to evaluate the RMST</li>
</ul>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">population</span> <span class="kw">&lt;-</span> <span class="fl">30</span> <span class="co"># For testing reasons it is set to a low number but ideally should be above 100            </span>
<span class="no">generations</span> <span class="kw">&lt;-</span><span class="fl">15</span> <span class="co"># For testing reasons it is set to a low number but ideally should be above 150            </span>
<span class="no">nCV</span> <span class="kw">&lt;-</span> <span class="fl">5</span>
<span class="no">distancetype</span> <span class="kw">&lt;-</span> <span class="st">"pearson"</span>
<span class="no">TournamentSize</span> <span class="kw">&lt;-</span> <span class="fl">2</span>
<span class="no">period</span> <span class="kw">&lt;-</span> <span class="fl">3650</span></pre></body></html></div>
</div>
<div id="run-galgo-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#run-galgo-algorithm" class="anchor"></a>Run Galgo algorithm</h3>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">264</span>)
<span class="no">output</span> <span class="kw">&lt;-</span> <span class="kw pkg">galgoR</span><span class="kw ns">::</span><span class="fu"><a href="../reference/galgo.html">galgo</a></span>(<span class="kw">generations</span> <span class="kw">=</span> <span class="no">generations</span>,
                        <span class="kw">population</span> <span class="kw">=</span> <span class="no">population</span>,
                        <span class="kw">prob_matrix</span> <span class="kw">=</span> <span class="no">train_expr</span>,
                        <span class="kw">OS</span> <span class="kw">=</span> <span class="no">train_surv</span>,
                        <span class="kw">nCV</span> <span class="kw">=</span> <span class="no">nCV</span>,
                        <span class="kw">distancetype</span> <span class="kw">=</span> <span class="no">distancetype</span>,
                        <span class="kw">TournamentSize</span> <span class="kw">=</span> <span class="no">TournamentSize</span>,
                        <span class="kw">period</span> <span class="kw">=</span> <span class="no">period</span>)</pre></body></html></div>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="no">output</span>))
<span class="co">#&gt; [1] "galgo.Obj"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "galgoR"</span></pre></body></html></div>
</div>
<div id="galgo-object" class="section level3">
<h3 class="hasAnchor">
<a href="#galgo-object" class="anchor"></a>Galgo Object</h3>
<p>The output of the galgo() function is an object of type <code>galgo.Obj</code> that has two slots with the elements:</p>
<ul>
<li>Solutions</li>
<li>ParetoFront.</li>
</ul>
<div id="solutions" class="section level4">
<h4 class="hasAnchor">
<a href="#solutions" class="anchor"></a>Solutions</h4>
<p>Is a l x (n + 5) matrix where n is the number of features evaluated and l is the number of solutions obtained.</p>
<ul>
<li>The submatrix l x n is a binary matrix where each row represents the chromosome of an evolved solution from the solution population, where each feature can be present (1) or absent (0) in the solution.</li>
<li>Column n+1 represent the k number of clusters for each solutions</li>
<li>Column n+2 shows the SC Fitness</li>
<li>Column n+3 represent Survival Fitness values</li>
<li>Column n+4 shows the solution rank</li>
<li>Column n+5 represent the crowding distance of the solution in the final pareto front</li>
</ul>
</div>
<div id="paretofront" class="section level4">
<h4 class="hasAnchor">
<a href="#paretofront" class="anchor"></a>ParetoFront</h4>
<p>Is a list of length equal to the number of generations run in the algorithm. Each element is a l x 2 matrix where l is the number of solutions obtained and the columns are the SC Fitness and the Survival Fitness values respectively.</p>
<p>For easier interpretation of the <code>galgo.Obj</code>, the output can be transformed to a List or to a DataFrame objects.</p>
</div>
</div>
</div>
<div id="to_list-function" class="section level2">
<h2 class="hasAnchor">
<a href="#to_list-function" class="anchor"></a>to_list() function</h2>
<p>This function restructurates a <code>galgo.Obj</code> to a more easy to understand an use list. This output is particularly useful if one wants to select a given solution and use its outputs in a new classifier. The output of type list has a length equals to the number of solutions obtained by the galgo algorithm.</p>
<p>Basically this output is a list of lists, where each element of the output is named after the solution’s name (solution.n, where n is the number assigned to that solution), and inside of it, it has all the constituents for that given solution with the following structure:</p>
<ul>
<li>solution.n$Genes: A vector of the features included in the solution</li>
<li>solution.n$k: The number of partitions found in that solution</li>
<li>solution.n$SC.Fit: The average silhouette coefficient of the partitions found</li>
<li>solution.n$Surv.Fit: The survival fitnes value</li>
<li>solution.n$Rank: The solution rank</li>
<li>CrowD: The solution crowding distance related to the rest of the solutions</li>
</ul>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">outputList</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/to_list.html">to_list</a></span>(<span class="no">output</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">outputList</span>))
<span class="co">#&gt; [1] "Solution.1" "Solution.2" "Solution.3" "Solution.4" "Solution.5"</span>
<span class="co">#&gt; [6] "Solution.6"</span></pre></body></html></div>
<p>To evaluate the structure of the first solution we can run:</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">outputList</span><span class="kw">[[</span><span class="st">"Solution.1"</span>]]
<span class="co">#&gt; $Genes</span>
<span class="co">#&gt;  [1] "KRT5"         "RRM2"         "MYC"          "CDC20"        "PTTG1"       </span>
<span class="co">#&gt;  [6] "CDC6"         "KIF2C"        "FGFR4"        "CCNB1"        "FOXA1"       </span>
<span class="co">#&gt; [11] "MDM2"         "NOL11"        "ANKRD2"       "DNAJB4"       "EPN3"        </span>
<span class="co">#&gt; [16] "KRT81"        "ADORA2B"      "NFATC2IP"     "DHTKD1"       "C7orf16"     </span>
<span class="co">#&gt; [21] "BYSL"         "PILRA"        "CDK20"        "VPS8"         "ZNF177"      </span>
<span class="co">#&gt; [26] "CD6"          "BSDC1"        "BTD"          "LTBP2"        "RABEP2"      </span>
<span class="co">#&gt; [31] "COX6CP1"      "GOLGA8C"      "SLIT3"        "PPP2R5E"      "ZNF638"      </span>
<span class="co">#&gt; [36] "GALNS"        "MST1"         "NGF"          "PRKCB"        "NDUFS6"      </span>
<span class="co">#&gt; [41] "VPS33A"       "PPFIBP1"      "GIF"          "STK17A"       "LOC100130134"</span>
<span class="co">#&gt; [46] "GPR63"        "RFC2"         "DEC1"         "ELF3"        </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $k</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $SC.Fit</span>
<span class="co">#&gt; [1] 0.2130412</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Surv.Fit</span>
<span class="co">#&gt; [1] 582.7308</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $rank</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $CrowD</span>
<span class="co">#&gt; [1] Inf</span></pre></body></html></div>
</div>
<div id="to_dataframe-function" class="section level2">
<h2 class="hasAnchor">
<a href="#to_dataframe-function" class="anchor"></a>to_dataframe() function</h2>
<p>The current function restructurates a <code>galgo.Obj</code> to a more easy to understand an use <code>data.frame</code>. The output data frame has m x n dimensions, were the rownames (m) are the solutions obtained by the galgo algorithm. The columns has the following structure:</p>
<ul>
<li>Genes: The features included in each solution in form of a list</li>
<li>k: The number of partitions found in that solution</li>
<li>SC.Fit: The average silhouette coefficient of the partitions found</li>
<li>Surv.Fit: The survival fitnes value</li>
<li>Rank: The solution rank</li>
<li>CrowD: The solution crowding distance related to the rest of the solutions</li>
</ul>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">outputDF</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/to_dataframe.html">to_dataframe</a></span>(<span class="no">output</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">outputDF</span>)
<span class="co">#&gt;                    Genes k     SC.Fit  Surv.Fit Rank     CrowD</span>
<span class="co">#&gt; Solutions.1 KRT5, RR.... 2 0.21304116  582.7308    1       Inf</span>
<span class="co">#&gt; Solutions.2 KRT5, RR.... 6 0.06620207 1004.2343    1       Inf</span>
<span class="co">#&gt; Solutions.3 MYBL2, K.... 2 0.17975311  611.6798    1 0.6759036</span>
<span class="co">#&gt; Solutions.4 MYBL2, B.... 3 0.09873938  680.6145    1 0.5871421</span>
<span class="co">#&gt; Solutions.5 MYBL2, B.... 4 0.08412215  714.7944    1 0.4940855</span>
<span class="co">#&gt; Solutions.6 MYBL2, R.... 2 0.20860066  408.4265    2       Inf</span></pre></body></html></div>
</div>
<div id="plot_pareto" class="section level2">
<h2 class="hasAnchor">
<a href="#plot_pareto" class="anchor"></a>plot_pareto()</h2>
<p>Once we obtain the <code>galgo.obj</code> from the output of <code><a href="../reference/galgo.html">galgo()</a></code> we can plot the obtained Pareto front and see how it evolved trough the tested number of generations</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_pareto.html">plot_pareto</a></span>(<span class="no">output</span>)</pre></body></html></div>
<p><img src="galgoR_files/figure-html/plot_pareto-1.png" width="700"></p>
</div>
</div>
<div id="case-study" class="section level1">
<h1 class="hasAnchor">
<a href="#case-study" class="anchor"></a>Case study</h1>
<p>Breast cancer (BRCA) is the most common neoplasm in women to date and one of the best studied cnacer types. Currently, numerous molecular alteration for this type of cancer are well known and many transcriptomic signatures have been developed for this type of cancer. In this regards, <a href="XXX">Perou et al.</a> proposed one of the first molecular subtype classification according to transcriptomic profiles of the tumor, which recapitulates naturally-occurring gene expression patterns that encompass different functional pathways and patient outcomes. These subtypes, (LumA, LumB, Basal-like, HER2 and Normal-Like) have a strong overap with the classical histopathological classification of BRCA tumors and migh affect decission making when used to decided chemotherapy in certain cases.</p>
<div id="data-preprocessing" class="section level2">
<h2 class="hasAnchor">
<a href="#data-preprocessing" class="anchor"></a>Data Preprocessing</h2>
<p>To evaluate Galgo’s performance along with PAM50 classification, we will use the two already scaled and reduced BRCA gene expression datasets and will compare Galgo performance with the widely used intrinsic molecular subtype PAM50 classification. Galgo performs feature selection by design, so this step is not strictly necessary to use galgoR (although feature selection might fasten GalgoR runs), nevertheless, appropiate gene expression scaling is critical when runing galgoR.</p>
</div>
<div id="breast-cancer-classification" class="section level2">
<h2 class="hasAnchor">
<a href="#breast-cancer-classification" class="anchor"></a>Breast cancer classification</h2>
<p>The scaled expression values of each patient are compared with the prototypical centroids using Pearson’s correlation coefficient and the closest centroid to each patient is used to assign the corresponding labels.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="co">#The reduced UPP dataset will be used as training set </span>
<span class="no">train_expression</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Train</span>)
<span class="no">train_clinic</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Train</span>)
<span class="no">train_features</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">Train</span>)
<span class="no">train_surv</span><span class="kw">&lt;-</span> <span class="kw pkg">survival</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">train_clinic</span>$<span class="no">t.rfs</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">train_clinic</span>$<span class="no">e.rfs</span>)

<span class="co">#The reduced TRANSBIG dataset will be used as test set </span>

<span class="no">test_expression</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">Test</span>)
<span class="no">test_clinic</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">Test</span>)
<span class="no">test_features</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">Test</span>)
<span class="no">test_surv</span><span class="kw">&lt;-</span> <span class="kw pkg">survival</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">test_clinic</span>$<span class="no">t.rfs</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">test_clinic</span>$<span class="no">e.rfs</span>)


<span class="co">#PAM50 centroids</span>
<span class="no">centroids</span><span class="kw">&lt;-</span> <span class="kw pkg">genefu</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/genefu/man/pam50.html">pam50</a></span>$<span class="no">centroids</span>
<span class="co">#Extract features from both data.frames</span>
<span class="no">inBoth</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html">Reduce</a></span>(<span class="no">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">train_expression</span>),<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">centroids</span>)))

<span class="co">#Classify samples </span>

<span class="no">PAM50_train</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/cluster_classify.html">cluster_classify</a></span>(<span class="no">train_expression</span>[<span class="no">inBoth</span>,],<span class="no">centroids</span>[<span class="no">inBoth</span>,],<span class="kw">method</span> <span class="kw">=</span> <span class="st">"spearman"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">PAM50_train</span>)
<span class="co">#&gt; PAM50_train</span>
<span class="co">#&gt;  1  2  3  4  5 </span>
<span class="co">#&gt; 22 30 94 73 15</span>

<span class="no">PAM50_test</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/cluster_classify.html">cluster_classify</a></span>(<span class="no">test_expression</span>[<span class="no">inBoth</span>,],<span class="no">centroids</span>[<span class="no">inBoth</span>,],<span class="kw">method</span> <span class="kw">=</span> <span class="st">"spearman"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">PAM50_test</span>)
<span class="co">#&gt; PAM50_test</span>
<span class="co">#&gt;  1  2  3  4  5 </span>
<span class="co">#&gt; 45 26 80 44  3</span>

<span class="co"># Classify samples using genefu</span>
<span class="co">#annot&lt;- fData(Train)</span>
<span class="co">#colnames(annot)[3]="Gene.Symbol"</span>
<span class="co">#PAM50_train&lt;- molecular.subtyping(sbt.model = "pam50",data = t(train_expression), annot = annot,do.mapping = TRUE)</span></pre></body></html></div>
<p>Once the patients are classified according to their closest centroids, we can now evaluate the survival curves for the different types in each of the datasets</p>
<div id="survival-of-upp-patients" class="section level3">
<h3 class="hasAnchor">
<a href="#survival-of-upp-patients" class="anchor"></a>Survival of UPP patients</h3>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(train_clinic$t.rfs,train_clinic$e.rfs)~ PAM50_train"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"UPP breast cancer \n PAM50 subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">3650</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="galgoR_files/figure-html/pam50_surv_UPP-1.png" width="700"></p>
</div>
<div id="survival-of-transbig-patients" class="section level3">
<h3 class="hasAnchor">
<a href="#survival-of-transbig-patients" class="anchor"></a>Survival of TRANSBIG patients</h3>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$t.rfs,test_clinic$e.rfs)~ PAM50_test"</span>)
<span class="no">tumortotal2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal2diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal2pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal2diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal2diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal2</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"TRANSBIG breast cancer \n PAM50 subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">3650</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="galgoR_files/figure-html/pam50_surv_TRANSBIG-1.png" width="700"></p>
</div>
</div>
<div id="find-breast-cancer-gene-signatures-with-galgor" class="section level2">
<h2 class="hasAnchor">
<a href="#find-breast-cancer-gene-signatures-with-galgor" class="anchor"></a>Find breast cancer gene signatures with GalgoR</h2>
<p>Now we run Galgo to find cohesive and clinically meaningful signatures for BRCA using UPP data as training set and TRANSBIG data as test set</p>
<div id="set-configuration-parameters" class="section level3">
<h3 class="hasAnchor">
<a href="#set-configuration-parameters" class="anchor"></a>Set configuration parameters</h3>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">population</span> <span class="kw">&lt;-</span> <span class="fl">15</span>
<span class="no">generations</span> <span class="kw">&lt;-</span><span class="fl">5</span>
<span class="no">nCV</span> <span class="kw">&lt;-</span> <span class="fl">5</span>
<span class="no">distancetype</span> <span class="kw">&lt;-</span> <span class="st">"pearson"</span>
<span class="no">TournamentSize</span> <span class="kw">&lt;-</span> <span class="fl">2</span>
<span class="no">period</span> <span class="kw">&lt;-</span> <span class="fl">3650</span>
<span class="no">usegpu</span> <span class="kw">&lt;-</span> <span class="fl">FALSE</span></pre></body></html></div>
<p>Run Galgo on the training set</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">output</span><span class="kw">=</span> <span class="kw pkg">galgoR</span><span class="kw ns">::</span><span class="fu"><a href="../reference/galgo.html">galgo</a></span>(<span class="kw">generations</span> <span class="kw">=</span> <span class="no">generations</span>, <span class="kw">population</span> <span class="kw">=</span> <span class="no">population</span>,<span class="kw">prob_matrix</span> <span class="kw">=</span> <span class="no">train_expression</span>, <span class="kw">OS</span><span class="kw">=</span><span class="no">train_surv</span>,<span class="kw">usegpu</span><span class="kw">=</span><span class="no">usegpu</span>,<span class="kw">nCV</span><span class="kw">=</span> <span class="no">nCV</span>, <span class="kw">distancetype</span><span class="kw">=</span><span class="no">distancetype</span>, <span class="kw">TournamentSize</span><span class="kw">=</span><span class="no">TournamentSize</span>, <span class="kw">period</span><span class="kw">=</span><span class="no">period</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="no">output</span>))</pre></body></html></div>
</div>
</div>
<div id="analyzing-galgo-results" class="section level2">
<h2 class="hasAnchor">
<a href="#analyzing-galgo-results" class="anchor"></a>Analyzing Galgo results</h2>
<div id="pareto-front" class="section level3">
<h3 class="hasAnchor">
<a href="#pareto-front" class="anchor"></a>Pareto front</h3>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_pareto.html">plot_pareto</a></span>(<span class="no">output</span>)</pre></body></html></div>
<p><img src="galgoR_files/figure-html/pareto_2-1.png" width="100%"></p>
</div>
<div id="summary-of-the-results" class="section level3">
<h3 class="hasAnchor">
<a href="#summary-of-the-results" class="anchor"></a>Summary of the results</h3>
<div class="sourceCode" id="cb25"><html><body><pre class="r">
<span class="no">output_df</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/to_dataframe.html">to_dataframe</a></span>(<span class="no">output</span>)

<span class="no">NonDom_solutions</span><span class="kw">&lt;-</span> <span class="no">output_df</span>[<span class="no">output_df</span>$<span class="no">Rank</span><span class="kw">==</span><span class="fl">1</span>,]

<span class="co"># N of non-dominated solutions </span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">NonDom_solutions</span>)
<span class="co">#&gt; [1] 4</span>

<span class="co"># N of partitions found</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">k</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; 2 5 9 </span>
<span class="co">#&gt; 2 1 1</span>

<span class="co">#Average N of genes per signature</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">Genes</span>,<span class="no">length</span>)))
<span class="co">#&gt; [1] 99.75</span>

<span class="co">#SC range</span>
<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">SC.Fit</span>)
<span class="co">#&gt; [1] 0.03119652 0.19908005</span>

<span class="co"># Survival fitnesss range</span>
<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">Surv.Fit</span>)
<span class="co">#&gt; [1] 444.2236 732.2119</span></pre></body></html></div>
<p>###Select best performing solutions</p>
<p>Now we select the best performing solutions for each number of partitions (k) according to C.Index</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r">
<span class="no">RESULT</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/non_dominated_summary.html">non_dominated_summary</a></span>(<span class="kw">output</span><span class="kw">=</span><span class="no">output</span>,<span class="kw">OS</span><span class="kw">=</span><span class="no">train_surv</span>, <span class="kw">prob_matrix</span><span class="kw">=</span> <span class="no">train_expression</span>, <span class="kw">distancetype</span> <span class="kw">=</span><span class="no">distancetype</span>, <span class="kw">usegpu</span><span class="kw">=</span> <span class="no">usegpu</span> )


<span class="no">best_sol</span><span class="kw">=</span><span class="kw">NULL</span>
  <span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">RESULT</span>$<span class="no">k</span>)){
  <span class="no">best_sol</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">best_sol</span>,<span class="no">RESULT</span>[<span class="no">RESULT</span>$<span class="no">k</span><span class="kw">==</span><span class="no">i</span>,<span class="st">"solution"</span>][<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="no">RESULT</span>[<span class="no">RESULT</span>$<span class="no">k</span><span class="kw">==</span><span class="no">i</span>,<span class="st">"C.Index"</span>])])
  }

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">best_sol</span>)
<span class="co">#&gt; [1] "Solutions.1" "Solutions.3" "Solutions.4"</span></pre></body></html></div>
</div>
<div id="create-prototypic-centroids" class="section level3">
<h3 class="hasAnchor">
<a href="#create-prototypic-centroids" class="anchor"></a>Create prototypic centroids</h3>
<p>Now we create the prototypic centroids of the selected solutions</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r">  <span class="no">CentroidsList</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create_centroids.html">create_centroids</a></span>(<span class="no">output</span>, <span class="kw">solution_names</span> <span class="kw">=</span> <span class="no">best_sol</span>, <span class="kw">trainset</span> <span class="kw">=</span> <span class="no">train_expression</span>)</pre></body></html></div>
</div>
</div>
<div id="test-galgo-signatures-in-a-test-set" class="section level2">
<h2 class="hasAnchor">
<a href="#test-galgo-signatures-in-a-test-set" class="anchor"></a>Test Galgo signatures in a test set</h2>
<p>We will test the Galgo signatures found with the UPP training set in an independent test set (TRANSBIG)</p>
<div id="classify-train-and-test-set-into-galgor-subtypes" class="section level3">
<h3 class="hasAnchor">
<a href="#classify-train-and-test-set-into-galgor-subtypes" class="anchor"></a>Classify train and test set into galgoR subtypes</h3>
<div class="sourceCode" id="cb28"><html><body><pre class="r">
<span class="no">train_classes</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/classify_multiple.html">classify_multiple</a></span>(<span class="kw">prob_matrix</span><span class="kw">=</span><span class="no">train_expression</span>,<span class="kw">centroid_list</span><span class="kw">=</span> <span class="no">CentroidsList</span>, <span class="kw">distancetype</span> <span class="kw">=</span> <span class="no">distancetype</span>)


<span class="no">test_classes</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/classify_multiple.html">classify_multiple</a></span>(<span class="kw">prob_matrix</span><span class="kw">=</span><span class="no">test_expression</span>,<span class="kw">centroid_list</span><span class="kw">=</span> <span class="no">CentroidsList</span>, <span class="kw">distancetype</span> <span class="kw">=</span> <span class="no">distancetype</span>)</pre></body></html></div>
</div>
<div id="calculate-train-and-test-set-c-index" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-train-and-test-set-c-index" class="anchor"></a>Calculate train and test set C.Index</h3>
<p>To calculate the train and test C.Index, the risk coefficients are calculated for each subclass in the training set and then are used to predict the risk of the different groups in the test set. This is particularly important for signatures with high number of partitions, were the survival differences of different groups might overlap and change their relative order, which is of great importance in the C.Index calculation.</p>
<div class="sourceCode" id="cb29"><html><body><pre class="r">
<span class="no">Prediction.models</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="no">best_sol</span>){

  <span class="no">OS</span><span class="kw">&lt;-</span> <span class="no">train_surv</span>
  <span class="no">predicted_class</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">train_classes</span>[,<span class="no">i</span>])
  <span class="no">predicted_classdf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">predicted_class</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">predicted_classdf</span>)<span class="kw">&lt;-</span> <span class="no">i</span>

  <span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"OS~ "</span>,<span class="no">i</span>))
  <span class="no">coxsimple</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/coxph.html">coxph</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">predicted_classdf</span>)


<span class="no">Prediction.models</span><span class="kw">[[</span><span class="no">i</span>]]<span class="kw">&lt;-</span> <span class="no">coxsimple</span>

}</pre></body></html></div>
</div>
<div id="calculate-c-index-for-training-and-test-set-using-the-prediction-models-of-the-training-set" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-c-index-for-training-and-test-set-using-the-prediction-models-of-the-training-set" class="anchor"></a>Calculate C.Index for training and test set using the prediction models of the training set</h3>
<div class="sourceCode" id="cb30"><html><body><pre class="r">
<span class="no">C.indexes</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">train_CI</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">NA</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">best_sol</span>)),<span class="kw">test_CI</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">NA</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">best_sol</span>)))
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">C.indexes</span>)<span class="kw">&lt;-</span> <span class="no">best_sol</span>

<span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="no">best_sol</span>){

<span class="no">predicted_class_train</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">train_classes</span>[,<span class="no">i</span>])
    <span class="no">predicted_class_train_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">predicted_class_train</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">predicted_class_train_df</span>)<span class="kw">&lt;-</span> <span class="no">i</span>

<span class="no">CI_train</span><span class="kw">&lt;-</span> <span class="fu">concordance.index</span>(<span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span>(<span class="no">Prediction.models</span><span class="kw">[[</span><span class="no">i</span>]],<span class="no">predicted_class_train_df</span>),<span class="kw">surv.time</span><span class="kw">=</span><span class="no">train_surv</span>[,<span class="fl">1</span>],<span class="kw">surv.event</span><span class="kw">=</span><span class="no">train_surv</span>[,<span class="fl">2</span>],<span class="kw">outx</span><span class="kw">=</span><span class="fl">FALSE</span>)$<span class="no">c.index</span>

<span class="no">C.indexes</span>[<span class="no">i</span>,<span class="st">"train_CI"</span>]<span class="kw">&lt;-</span> <span class="no">CI_train</span>

<span class="no">predicted_class_test</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">test_classes</span>[,<span class="no">i</span>])
    <span class="no">predicted_class_test_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">predicted_class_test</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">predicted_class_test_df</span>)<span class="kw">&lt;-</span> <span class="no">i</span>


<span class="no">CI_test</span><span class="kw">&lt;-</span> <span class="fu">concordance.index</span>(<span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span>(<span class="no">Prediction.models</span><span class="kw">[[</span><span class="no">i</span>]],<span class="no">predicted_class_test_df</span>),<span class="kw">surv.time</span><span class="kw">=</span><span class="no">test_surv</span>[,<span class="fl">1</span>],<span class="kw">surv.event</span><span class="kw">=</span><span class="no">test_surv</span>[,<span class="fl">2</span>],<span class="kw">outx</span><span class="kw">=</span><span class="fl">FALSE</span>)$<span class="no">c.index</span>

<span class="no">C.indexes</span>[<span class="no">i</span>,<span class="st">"test_CI"</span>]<span class="kw">&lt;-</span> <span class="no">CI_test</span>

}

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">C.indexes</span>)
<span class="co">#&gt;              train_CI   test_CI</span>
<span class="co">#&gt; Solutions.1 0.6527876 0.5489152</span>
<span class="co">#&gt; Solutions.3 0.6221393 0.5691913</span>
<span class="co">#&gt; Solutions.4 0.6200222 0.5850493</span>


<span class="no">best_signature</span><span class="kw">&lt;-</span> <span class="no">best_sol</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="no">C.indexes</span>$<span class="no">test_CI</span>)]

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">best_signature</span>)
<span class="co">#&gt; [1] "Solutions.4"</span></pre></body></html></div>
</div>
<div id="evaluate-prediction-survival-of-galgo-signatures" class="section level3">
<h3 class="hasAnchor">
<a href="#evaluate-prediction-survival-of-galgo-signatures" class="anchor"></a>Evaluate prediction survival of Galgo signatures</h3>
<p>We test best galgo signature with training and test sets</p>
<div class="sourceCode" id="cb31"><html><body><pre class="r">
<span class="no">train_class</span> <span class="kw">&lt;-</span> <span class="no">train_classes</span>[,<span class="no">best_signature</span>]

<span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(train_clinic$t.rfs,train_clinic$e.rfs)~ train_class"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"UPP breast cancer \n Galgo subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">3650</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="galgoR_files/figure-html/galgo_train_surv-1.png" width="100%"></p>
<div class="sourceCode" id="cb32"><html><body><pre class="r">
<span class="no">test_class</span> <span class="kw">&lt;-</span> <span class="no">test_classes</span>[,<span class="no">best_signature</span>]

<span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$t.rfs,test_clinic$e.rfs)~ test_class"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"TRANSBIG breast cancer \n Galgo subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">3650</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="galgoR_files/figure-html/galgo_test_surv-1.png" width="100%"></p>
</div>
</div>
<div id="comparison-of-galgo-vs-pam50-classifier" class="section level2">
<h2 class="hasAnchor">
<a href="#comparison-of-galgo-vs-pam50-classifier" class="anchor"></a>Comparison of Galgo vs PAM50 classifier</h2>
<p>Compare PAM50 classification vs Galgo classification in the TRANSBIG (test) dataset</p>
<div class="sourceCode" id="cb33"><html><body><pre class="r">

<span class="no">surv_formula1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$t.rfs,test_clinic$e.rfs)~ test_class"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula1</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="no">surv_formula2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$t.rfs,test_clinic$e.rfs)~ PAM50_test"</span>)
<span class="no">tumortotal2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula2</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal2diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula2</span>)
<span class="no">tumortotal2pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal2diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)




  <span class="no">SURV</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">GALGO</span><span class="kw">=</span><span class="no">tumortotal1</span>,<span class="kw">PAM50</span><span class="kw">=</span><span class="no">tumortotal2</span> )
  <span class="no">COLS</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>:<span class="fl">8</span>,<span class="fl">10</span>)
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">cex</span><span class="kw">=</span><span class="fl">1.35</span>, <span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">3.8</span>, <span class="fl">3.8</span>, <span class="fl">2.5</span>, <span class="fl">2.5</span>) + <span class="fl">0.1</span>)
  <span class="no">p</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">SURV</span>,<span class="kw">combine</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"Galgo vs. PAM50 subtypes \n BRCA survival comparison"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="no">period</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="galgoR_files/figure-html/test_pam50-1.png" width="100%"></p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 4.0.0 (2020-04-24)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 18.04.4 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] survminer_0.4.7             ggpubr_0.3.0               </span>
<span class="co">#&gt;  [3] ggplot2_3.3.1               genefu_2.21.0              </span>
<span class="co">#&gt;  [5] AIMS_1.21.0                 e1071_1.7-3                </span>
<span class="co">#&gt;  [7] iC10_1.5                    iC10TrainingData_1.3.1     </span>
<span class="co">#&gt;  [9] impute_1.63.0               pamr_1.56.1                </span>
<span class="co">#&gt; [11] cluster_2.1.0               biomaRt_2.45.0             </span>
<span class="co">#&gt; [13] limma_3.45.4                mclust_5.4.6               </span>
<span class="co">#&gt; [15] survcomp_1.39.0             prodlim_2019.11.13         </span>
<span class="co">#&gt; [17] survival_3.1-12             Biobase_2.49.0             </span>
<span class="co">#&gt; [19] BiocGenerics_0.35.4         galgoR_0.4.0               </span>
<span class="co">#&gt; [21] breastCancerUPP_1.27.0      breastCancerTRANSBIG_1.27.0</span>
<span class="co">#&gt; [23] BiocStyle_2.17.0           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] amap_0.8-18          colorspace_1.4-1     ggsignif_0.6.0      </span>
<span class="co">#&gt;   [4] ellipsis_0.3.1       class_7.3-17         rio_0.5.16          </span>
<span class="co">#&gt;   [7] rprojroot_1.3-2      fs_1.4.1             rstudioapi_0.11     </span>
<span class="co">#&gt;  [10] farver_2.0.3         bit64_0.9-7          AnnotationDbi_1.51.0</span>
<span class="co">#&gt;  [13] codetools_0.2-16     splines_4.0.0        doParallel_1.0.15   </span>
<span class="co">#&gt;  [16] knitr_1.28           SuppDists_1.1-9.5    km.ci_0.5-2         </span>
<span class="co">#&gt;  [19] broom_0.5.6          dbplyr_1.4.4         BiocManager_1.30.10 </span>
<span class="co">#&gt;  [22] compiler_4.0.0       httr_1.4.1           backports_1.1.7     </span>
<span class="co">#&gt;  [25] assertthat_0.2.1     Matrix_1.2-18        htmltools_0.4.0     </span>
<span class="co">#&gt;  [28] prettyunits_1.1.1    tools_4.0.0          gtable_0.3.0        </span>
<span class="co">#&gt;  [31] glue_1.4.1           dplyr_1.0.0          rappdirs_0.3.1      </span>
<span class="co">#&gt;  [34] Rcpp_1.0.4.6         carData_3.0-4        cellranger_1.1.0    </span>
<span class="co">#&gt;  [37] pkgdown_1.5.1        vctrs_0.3.1          nlme_3.1-147        </span>
<span class="co">#&gt;  [40] iterators_1.0.12     xfun_0.14            stringr_1.4.0       </span>
<span class="co">#&gt;  [43] openxlsx_4.1.5       lifecycle_0.2.0      rstatix_0.5.0       </span>
<span class="co">#&gt;  [46] XML_3.99-0.3         zoo_1.8-8            MASS_7.3-51.6       </span>
<span class="co">#&gt;  [49] scales_1.1.1         hms_0.5.3            yaml_2.2.1          </span>
<span class="co">#&gt;  [52] curl_4.3             gridExtra_2.3        memoise_1.1.0       </span>
<span class="co">#&gt;  [55] KMsurv_0.1-5         stringi_1.4.6        RSQLite_2.2.0       </span>
<span class="co">#&gt;  [58] rmeta_3.0            S4Vectors_0.27.11    desc_1.2.0          </span>
<span class="co">#&gt;  [61] foreach_1.5.0        zip_2.0.4            lava_1.6.7          </span>
<span class="co">#&gt;  [64] rlang_0.4.6          pkgconfig_2.0.3      evaluate_0.14       </span>
<span class="co">#&gt;  [67] lattice_0.20-41      purrr_0.3.4          labeling_0.3        </span>
<span class="co">#&gt;  [70] survivalROC_1.0.3    bit_1.1-15.2         tidyselect_1.1.0    </span>
<span class="co">#&gt;  [73] magrittr_1.5         bookdown_0.19        R6_2.4.1            </span>
<span class="co">#&gt;  [76] IRanges_2.23.8       generics_0.0.2       bootstrap_2019.6    </span>
<span class="co">#&gt;  [79] DBI_1.1.0            pillar_1.4.4         haven_2.3.1         </span>
<span class="co">#&gt;  [82] foreign_0.8-79       withr_2.2.0          abind_1.4-5         </span>
<span class="co">#&gt;  [85] tibble_3.0.1         crayon_1.3.4         car_3.0-8           </span>
<span class="co">#&gt;  [88] survMisc_0.5.5       KernSmooth_2.23-17   BiocFileCache_1.13.0</span>
<span class="co">#&gt;  [91] rmarkdown_2.2        progress_1.2.2       grid_4.0.0          </span>
<span class="co">#&gt;  [94] readxl_1.3.1         data.table_1.12.8    blob_1.2.1          </span>
<span class="co">#&gt;  [97] forcats_0.5.0        digest_0.6.25        xtable_1.8-4        </span>
<span class="co">#&gt; [100] tidyr_1.1.0          openssl_1.4.1        stats4_4.0.0        </span>
<span class="co">#&gt; [103] munsell_0.5.0        nsga2R_1.0           askpass_1.1</span></pre></body></html></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Martin Guerrero, Carlos Catania.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
