<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GalgoR user Guide • galgoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GalgoR user Guide">
<meta property="og:description" content="galgoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">galgoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Galgo_vignette.html">GalgoR user Guide</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/harpomaxx/galgo/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>GalgoR user Guide</h1>
                        <h4 class="author">Martin E. Guerrero-Gimenez</h4>
            <address class="author_afil">
      Laboratory of Oncology, Institute of Medicine and Experimental Biology of Cuyo (IMBECU), National Scientific and Technical Research Council (CONICET), Mendoza, Argentina.<br><a class="author_email" href="mailto:#"></a><a href="mailto:mguerrero@mendoza-conicet.gob.ar">mguerrero@mendoza-conicet.gob.ar</a>
      </address>
                              <h4 class="author">Juan Manuel Fernandez-Muñoz</h4>
            <address class="author_afil">
      Laboratory of Oncology, Institute of Medicine and Experimental Biology of Cuyo (IMBECU), National Scientific and Technical Research Council (CONICET), Mendoza, Argentina.<br><h4 class="author">Carlos A. Catania</h4>
            <address class="author_afil">
      Engineering School, National University of Cuyo, Mendoza, Argentina.<br><h4 class="date">2020-06-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/harpomaxx/galgo/blob/master/vignettes/Galgo_vignette.Rmd"><code>vignettes/Galgo_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>Galgo_vignette.Rmd</code></div>

    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>We report a novel method to identify specific transcriptomic phenotypes based on an elitist non-dominated sorting genetic algorithm that combines the advantages of clustering methods and the exploratory properties of genetic algorithms to discover biologically and clinically relevant molecular subtypes in different cancers.</p>
    </div>
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>In the new era of omics data, precision medicine has become the new paradigm of cancer treatment. Among all available omics techniques, gene expression profiling, in particular, has been increasingly used to classify tumor subtypes with different biological behavior. Cancer subtype discovery is usually approached from two possible perspectives:</p>
<ul>
<li>Using the molecular data alone with unsupervised techniques such as clustering analysis.</li>
<li>Using supervised techniques focusing entirely on survival data.</li>
</ul>
<p>The problem of finding patients subgroups with survival differences while maintaining cluster consistency could be viewed as a bi-objective problem, where there is a trade-off between the separability of the different groups and the ability of a given signature to consistently distinguish patients with different clinical outcomes. This gives rise to a set of optimal solutions, also known as Pareto-optimal solutions. To overcome these issues, we combined the advantages of clustering methods for grouping heterogeneous omics data and the search properties of genetic algorithms in galgoR: A flexible yet robust multi-objective meta-heuristic for disease subtype discovery based on an elitist non-dominated sorting genetic algorithm (NSGA-II), driven by the underlying premise of maximizing survival differences between groups while getting high consistency and robustness of the clusters obtained.</p>
</div>
<div id="algorithm" class="section level1">
<h1 class="hasAnchor">
<a href="#algorithm" class="anchor"></a>Algorithm</h1>
<p>In the galgoR package, the NSGA-II framework was used for finding multiple Pareto-optimal solutions to classify patients according to their gene expression patterns. Basically, NSGA-II starts with a population of competing individuals which are evaluated under a set of fitness functions that estimate the survival differences and cohesiveness of the different transcriptomic groups. Then, solutions are ranked and sorted according to their non-domination level which will affect the way they are chosen to be submitted to the so-called “evolutionary operators” such as crossover and mutation. Once a set of well-suited solutions are selected and reproduced, a new offspring of individuals composed of a mixture of the “genetic information” of the parents is obtained. Parents and offspring are pooled and the best-ranked solutions are selected and passed to the next generation which will start over the same process again.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div id="galgor-library" class="section level2">
<h2 class="hasAnchor">
<a href="#galgor-library" class="anchor"></a>galgoR library</h2>
<p>To install galgoR package, start R and enter:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"https://github.com/harpomaxx/galgo"</span>)

<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">galgoR</span>)</pre></body></html></div>
</div>
<div id="examples-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#examples-datasets" class="anchor"></a>Examples datasets</h2>
<p>To standardize the structure of genomic data, we use the <a href="https://www.bioconductor.org/packages/release/bioc/html/Biobase.html">ExpressionSet</a> structure for the examples given in this guide. The <code>ExpressionSet</code> objects can hold different types of data in a single structure but in this case we opted for using a simplified format to facilitate the example to those not familiar with the <a href="https://www.bioconductor.org/packages/release/bioc/html/Biobase.html">Biobase</a> package. The <code>ExpressionSet</code> objects are formed mainly by:</p>
<ul>
<li>A matrix of genetic expression, usually derived from microarray or RNAseq experiments.</li>
<li>Phenotypic data, where we find information on the samples (condition, status, treatment, survival, and other covariates).</li>
<li>Finally, these objects can also contain Annotations and feature Meta-data.</li>
</ul>
<p>To start testing galgoR, the package contains two reduced lung adenocarcinoma gene expression datasets (<a href="https://gdc.cancer.gov/about-data/publications/luad_2014">TCGA</a> and <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE68465">GSE68465</a>), that can be download using the function use_rna_luad(). Additionally, It also contains the <a href="http://cancer.unc.edu/nhayes/publications/adenocarcinoma.2012/wilkerson.2012.LAD.predictor.centroids.csv.zip">Wilkerson’s centroids</a> to perform lung adenocarcinoma sample classification.</p>
<div class="sourceCode"><html><body><pre class="r"> <span class="no">rna_luad</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/use_rna_luad.html">use_rna_luad</a></span>()
 <span class="no">TCGA</span><span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">TCGA</span> <span class="co">#Access TCGA dataset</span>
 <span class="no">GSE68465</span><span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">GSE68465</span> <span class="co">#Access GSE68465 dataset</span>

 <span class="co">#To access gene expression data</span>
 <span class="no">TCGA_expr</span><span class="kw">&lt;-</span> <span class="no">TCGA</span>$<span class="no">expression_data</span>

 <span class="co">#To access feature data</span>
 <span class="no">TCGA_features</span><span class="kw">&lt;-</span> <span class="no">TCGA</span>$<span class="no">feature_data</span>

 <span class="co">#To access clinical data</span>
 <span class="no">TCGA_clinic</span> <span class="kw">&lt;-</span> <span class="no">TCGA</span>$<span class="no">pheno_data</span>

 <span class="co">#To get wilkerson centroids</span>
 <span class="no">WilkCentroids</span> <span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">WilkCentroids</span></pre></body></html></div>
</div>
</div>
<div id="examples" class="section level1">
<h1 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h1>
<div id="loading-data" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-data" class="anchor"></a>Loading data</h2>
<div class="sourceCode"><html><body><pre class="r"><span class="no">rna_luad</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/use_rna_luad.html">use_rna_luad</a></span>()</pre></body></html></div>
<p>We will also load the <code>survival</code> library</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">survival</span>)</pre></body></html></div>
</div>
<div id="run-galgo" class="section level2">
<h2 class="hasAnchor">
<a href="#run-galgo" class="anchor"></a>Run galgo()</h2>
<p>The main function in this package is galgo(). It accepts an expression matrix and survival object to find robust gene expression signatures related to a given outcome. This function contains some parameters that can be modified, according to the characteristics of the analysis to be performed.</p>
<div id="setting-parameters" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-parameters" class="anchor"></a>Setting parameters</h3>
<p>The principal parameters are:</p>
<ul>
<li>population: a number indicating the number of solutions in the population of solutions that will be evolved</li>
<li>generations: a number indicating the number of iterations of the galgo algorithm</li>
<li>nCV: number of cross-validation sets</li>
<li>usegpu: logical default to FALSE, set to TRUE if you wish to use gpu computing (gpuR package must be properly installed and loaded)</li>
<li>distancetype: character, it can be ‘pearson’ (centered pearson), ‘uncentered’ (uncentered pearson), ‘spearman’ or ‘euclidean’</li>
<li>TournamentSize: a number indicating the size of the tournaments for the selection procedure</li>
<li>period: a number indicating the outcome period to evaluate the RMST</li>
</ul>
<div class="sourceCode"><html><body><pre class="r"><span class="no">population</span> <span class="kw">&lt;-</span> <span class="fl">30</span> <span class="co"># For testing reasons it is set to a low number but ideally should be above 100            </span>
<span class="no">generations</span> <span class="kw">&lt;-</span><span class="fl">15</span> <span class="co"># For testing reasons it is set to a low number but ideally should be above 150            </span>
<span class="no">nCV</span> <span class="kw">&lt;-</span> <span class="fl">5</span>
<span class="no">distancetype</span> <span class="kw">&lt;-</span> <span class="st">"pearson"</span>
<span class="no">TournamentSize</span> <span class="kw">&lt;-</span> <span class="fl">2</span>
<span class="no">period</span> <span class="kw">&lt;-</span> <span class="fl">1825</span></pre></body></html></div>
</div>
<div id="expression-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#expression-matrix" class="anchor"></a>Expression matrix</h3>
<p>Create an expression matrix for the TCGA_LUAD data example.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">TCGA_expr</span> <span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">TCGA</span>$<span class="no">expression_matrix</span></pre></body></html></div>
</div>
<div id="survival-object" class="section level3">
<h3 class="hasAnchor">
<a href="#survival-object" class="anchor"></a>Survival Object</h3>
<p>The ‘OS’ object is created by the Surv() function of the survival package. This uses phenotypic data that are contained in the TCGA dataset.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">TCGA_clinic</span> <span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">TCGA</span>$<span class="no">pheno_data</span>

<span class="no">OS</span> <span class="kw">&lt;-</span> <span class="kw pkg">survival</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">TCGA_clinic</span>$<span class="no">time</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">TCGA_clinic</span>$<span class="no">status</span>)</pre></body></html></div>
</div>
<div id="run-galgo-algorithm" class="section level3">
<h3 class="hasAnchor">
<a href="#run-galgo-algorithm" class="anchor"></a>Run Galgo algorithm</h3>
<div class="sourceCode"><html><body><pre class="r"><span class="no">output</span> <span class="kw">&lt;-</span> <span class="kw pkg">galgoR</span><span class="kw ns">::</span><span class="fu"><a href="../reference/galgo.html">galgo</a></span>(<span class="kw">generations</span> <span class="kw">=</span> <span class="no">generations</span>,
                        <span class="kw">population</span> <span class="kw">=</span> <span class="no">population</span>,
                        <span class="kw">prob_matrix</span> <span class="kw">=</span> <span class="no">TCGA_expr</span>,
                        <span class="kw">OS</span> <span class="kw">=</span> <span class="no">OS</span>,
                        <span class="kw">nCV</span> <span class="kw">=</span> <span class="no">nCV</span>,
                        <span class="kw">distancetype</span> <span class="kw">=</span> <span class="no">distancetype</span>,
                        <span class="kw">TournamentSize</span> <span class="kw">=</span> <span class="no">TournamentSize</span>,
                        <span class="kw">period</span> <span class="kw">=</span> <span class="no">period</span>)

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="no">output</span>))</pre></body></html></div>
<pre><code>#&gt; [1] "galgo.Obj"
#&gt; attr(,"package")
#&gt; [1] "galgoR"</code></pre>
</div>
<div id="galgo-object" class="section level3">
<h3 class="hasAnchor">
<a href="#galgo-object" class="anchor"></a>Galgo Object</h3>
<p>The output of the galgo() function is an object of type ‘galgo.Obj’ that has two slots with the elements:</p>
<ul>
<li>Solutions</li>
<li>ParetoFront.</li>
</ul>
<div id="solutions" class="section level4">
<h4 class="hasAnchor">
<a href="#solutions" class="anchor"></a>Solutions</h4>
<p>Is a l x (n + 5) matrix where n is the number of features evaluated and l is the number of solutions obtained.</p>
<ul>
<li>The submatrix l x n is a binary matrix where each row represents the chromosome of an evolved solution from the solution population, where each feature can be present (1) or absent (0) in the solution.</li>
<li>Column n+1 represent the k number of clusters for each solutions</li>
<li>Column n+2 shows the SC Fitness</li>
<li>Column n+3 represent Survival Fitness values</li>
<li>Column n+4 shows the solution rank</li>
<li>Column n+5 represent the crowding distance of the solution in the final pareto front</li>
</ul>
</div>
<div id="paretofront" class="section level4">
<h4 class="hasAnchor">
<a href="#paretofront" class="anchor"></a>ParetoFront</h4>
<p>Is a list of length equal to the number of generations run in the algorithm. Each element is a l x 2 matrix where l is the number of solutions obtained and the columns are the SC Fitness and the Survival Fitness values respectively.</p>
<p>For easier interpretation of the ‘galgo.Obj’, the output can be transformed to a List or to a DataFrame objects.</p>
</div>
</div>
</div>
<div id="tolist-function" class="section level2">
<h2 class="hasAnchor">
<a href="#tolist-function" class="anchor"></a>toList() function</h2>
<p>This function restructurates a galgo.Obj to a more easy to understand an use list. This output is particularly useful if one wants to select a given solution and use its outputs in a new classifier. The output of type list has a length equals to the number of solutions obtained by the galgo algorithm.</p>
<p>Basically this output is a list of lists, where each element of the output is named after the solution’s name (solution.n, where n is the number assigned to that solution), and inside of it, it has all the constituents for that given solution with the following structure:</p>
<ul>
<li>solution.n$Genes: A vector of the features included in the solution</li>
<li>solution.n$k: The number of partitions found in that solution</li>
<li>solution.n$SC.Fit: The average silhouette coefficient of the partitions found</li>
<li>solution.n$Surv.Fit: The survival fitnes value</li>
<li>solution.n$Rank: The solution rank</li>
<li>CrowD: The solution crowding distance related to the rest of the solutions</li>
</ul>
<div class="sourceCode"><html><body><pre class="r"><span class="no">outputList</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/toList.html">toList</a></span>(<span class="no">output</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">outputList</span>))
<span class="co">#&gt; [1] "Solution.1" "Solution.2" "Solution.3" "Solution.4" "Solution.5"</span>
<span class="co">#&gt; [6] "Solution.6"</span></pre></body></html></div>
<p>To evaluate the structure of the first solution we can run:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">outputList</span><span class="kw">[[</span><span class="st">"Solution.1"</span>]]
<span class="co">#&gt; $Genes</span>
<span class="co">#&gt;   [1] "ECT2"     "PBK"      "GINS1"    "H2AFZ"    "CLEC3B"   "CLDN18"  </span>
<span class="co">#&gt;   [7] "SNRPG"    "SEC23A"   "FBP1"     "NDUFS1"   "PRKDC"    "CD4"     </span>
<span class="co">#&gt;  [13] "FOLR2"    "ALOX5"    "LGALS9"   "LPXN"     "MUC4"     "PGC"     </span>
<span class="co">#&gt;  [19] "SFTPB"    "TDO2"     "TMEM45A"  "ADAMDEC1" "TMEM161A" "LRP6"    </span>
<span class="co">#&gt;  [25] "ANXA4"    "PLEC"     "PSMD9"    "LPIN1"    "ZNF606"   "TMEM184B"</span>
<span class="co">#&gt;  [31] "PORCN"    "SCNN1G"   "HNRNPH1"  "ZNF304"   "PDCL3"    "AGA"     </span>
<span class="co">#&gt;  [37] "FANCA"    "GAPDH"    "BAALC"    "EEF1B2"   "PIK3IP1"  "ALDH1L1" </span>
<span class="co">#&gt;  [43] "GNAI1"    "MTSS1L"   "AREL1"    "ASB4"     "MAPK6"    "OGDHL"   </span>
<span class="co">#&gt;  [49] "MTERF1"   "RWDD1"    "SLC25A40" "SNX7"     "MCM5"     "KIF16B"  </span>
<span class="co">#&gt;  [55] "RAC1"     "LRRC37A"  "ERCC3"    "MED31"    "WFDC1"    "BBS9"    </span>
<span class="co">#&gt;  [61] "TLR1"     "TNNT2"    "CCT8"     "ALPK3"    "RPS27"    "SLC39A8" </span>
<span class="co">#&gt;  [67] "SIM2"     "IL1RN"    "ADTRP"    "RMND5A"   "RAB11A"   "LARP7"   </span>
<span class="co">#&gt;  [73] "EGLN1"    "SPAG16"   "ATP10D"   "GM2A"     "RNF139"   "GTF2H3"  </span>
<span class="co">#&gt;  [79] "BASP1"    "FHL1"     "IFT22"    "RPSA"     "PPIF"     "YEATS4"  </span>
<span class="co">#&gt;  [85] "PTPN7"    "RRP9"     "MUTYH"    "KAT8"     "ECD"      "GPRC5D"  </span>
<span class="co">#&gt;  [91] "MPHOSPH9" "PRR5"     "TMEM70"   "KCTD7"    "SLC8A1"   "NOP2"    </span>
<span class="co">#&gt;  [97] "CCNT2"    "COQ8B"    "ITGA2"    "CPOX"     "PCDHB12"  "HSPB2"   </span>
<span class="co">#&gt; [103] "ETV5"     "BTK"      "RNF31"    "PWP1"     "B4GALNT1" "SELENOW" </span>
<span class="co">#&gt; [109] "INO80B"   "TESK2"   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $k</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $SC.Fit</span>
<span class="co">#&gt; [1] 0.1716594</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Surv.Fit</span>
<span class="co">#&gt; [1] 211.5115</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $rank</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $CrowD</span>
<span class="co">#&gt; [1] Inf</span></pre></body></html></div>
</div>
<div id="todataframe-function" class="section level2">
<h2 class="hasAnchor">
<a href="#todataframe-function" class="anchor"></a>toDataFrame() function</h2>
<p>The current function restructurates a galgo.Obj to a more easy to understand an use data.frame. The output data.frame has m x n dimensions, were the rownames (m) are the solutions obtained by the galgo algorithm. The columns has the following structure:</p>
<ul>
<li>Genes: The features included in each solution in form of a list</li>
<li>k: The number of partitions found in that solution</li>
<li>SC.Fit: The average silhouette coefficient of the partitions found</li>
<li>Surv.Fit: The survival fitnes value</li>
<li>Rank: The solution rank</li>
<li>CrowD: The solution crowding distance related to the rest of the solutions</li>
</ul>
<div class="sourceCode"><html><body><pre class="r"><span class="no">outputDF</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/toDataFrame.html">toDataFrame</a></span>(<span class="no">output</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">outputDF</span>)
<span class="co">#&gt;                    Genes  k    SC.Fit Surv.Fit Rank     CrowD</span>
<span class="co">#&gt; Solutions.1 ECT2, PB....  2 0.1716594 211.5115    1       Inf</span>
<span class="co">#&gt; Solutions.2 MAD2L1, .... 10 0.0248802 484.6104    1       Inf</span>
<span class="co">#&gt; Solutions.3 TOP2A, S....  3 0.1006130 310.2462    1 0.4842113</span>
<span class="co">#&gt; Solutions.4 PBK, GIN....  2 0.1663238 249.3958    1 0.3459966</span>
<span class="co">#&gt; Solutions.5 MAD2L1, ....  2 0.1282477 284.6221    1 0.3034071</span>
<span class="co">#&gt; Solutions.6 ECT2, PB....  5 0.0702194 335.8719    1 0.2925374</span></pre></body></html></div>
</div>
<div id="plot_pareto" class="section level2">
<h2 class="hasAnchor">
<a href="#plot_pareto" class="anchor"></a>plot_pareto()</h2>
<p>Once we obtain the <code>galgo.obj</code> from the output of <code><a href="../reference/galgo.html">galgo()</a></code> we can plot the obtained Pareto front and see how it evolved trough the tested number of generations</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_pareto.html">plot_pareto</a></span>(<span class="no">output</span>)</pre></body></html></div>
<p><img src="../inst/extdata/images/pareto1.jpg" width="100%"></p>
</div>
</div>
<div id="case-study" class="section level1">
<h1 class="hasAnchor">
<a href="#case-study" class="anchor"></a>Case study</h1>
<p>Lung adenocarcinoma (LUAD) is one of the most common types of cancer and, to date, still presents high mortality rates. Currently, numerous molecular alteration for this type of cancer are well known but, unlike breast cancer, very few transcriptomic signatures have been developed for this type of cancer. In this regards, <a href="https://pubmed.ncbi.nlm.nih.gov/22590557/">Wilkerson’s et al.</a> have proposed a transcriptomic classification into three different LUAD molecular subtypes (Bronchoid, Magnoid, and Squamoid) which recapitulates naturally-occurring gene expression patterns that encompass different functional pathways and patient outcomes.</p>
<div id="data-preprocessing" class="section level2">
<h2 class="hasAnchor">
<a href="#data-preprocessing" class="anchor"></a>Data Preprocessing</h2>
<p>To evaluate Galgo’s performance along with Wilkerson’s classification, we used two already scaled and reduced lung adenocarcinoma gene expression datasets included in the package (<a href="https://gdc.cancer.gov/about-data/publications/luad_2014">TCGA</a> and <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE68465">GSE68465</a>) that can be download using the function use_rna_luad(). Aditionally, It also contains the <a href="http://cancer.unc.edu/nhayes/publications/adenocarcinoma.2012/wilkerson.2012.LAD.predictor.centroids.csv.zip">Wilkerson’s centroids</a> to perform lung adenocarcinoma sample classification. <a href="http://cancer.unc.edu/nhayes/publications/adenocarcinoma.2012/wilkerson.2012.LAD.predictor.centroids.csv.zip">Wilkerson’s centroids</a> were used to classify samples according to their corresponding molecular subtype.</p>
</div>
<div id="lung-adenocarcinoma-classification" class="section level2">
<h2 class="hasAnchor">
<a href="#lung-adenocarcinoma-classification" class="anchor"></a>Lung adenocarcinoma classification</h2>
<p>The scaled expression values of each patient are compared with the prototypical centroids using Pearson’s correlation coefficient and the closest centroid to each patient is used to assign the corresponding labels.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">rna_luad</span><span class="kw">&lt;-</span><span class="fu"><a href="../reference/use_rna_luad.html">use_rna_luad</a></span>()

<span class="co">#The expression of the toy datasets are already scaled</span>

<span class="co">#The TCGA dataset will be used as training set </span>
<span class="no">train_expression</span> <span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">TCGA</span>$<span class="no">expression_matrix</span>
<span class="no">train_clinic</span><span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">TCGA</span>$<span class="no">pheno_data</span>
<span class="no">train_features</span><span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">TCGA</span>$<span class="no">feature_data</span>
<span class="no">train_surv</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">train_clinic</span>$<span class="no">time</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">train_clinic</span>$<span class="no">status</span>)

<span class="co">#The TCGA dataset will be used as test set </span>
<span class="no">test_expression</span> <span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">GSE68465</span>$<span class="no">expression_matrix</span>
<span class="no">test_clinic</span><span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">GSE68465</span>$<span class="no">pheno_data</span>
<span class="no">test_features</span><span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">GSE68465</span>$<span class="no">feature_data</span>
<span class="no">test_surv</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">test_clinic</span>$<span class="no">time</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">test_clinic</span>$<span class="no">status</span>)
<span class="co">#We change the rownames to be gene Symbol insted of Gene Id.</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">train_expression</span>)<span class="kw">&lt;-</span> <span class="no">train_features</span>$<span class="no">gene</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">test_expression</span>)<span class="kw">&lt;-</span> <span class="no">test_features</span>$<span class="no">gene</span>

<span class="co">#Wilkerson's centroids</span>
<span class="no">centroids</span><span class="kw">&lt;-</span> <span class="no">rna_luad</span>$<span class="no">WilkCentroids</span>

<span class="co">#Extract features from both data.frames</span>
<span class="no">inBoth</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html">Reduce</a></span>(<span class="no">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">train_expression</span>),<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">centroids</span>)))

<span class="co">#Classify samples</span>

<span class="no">Wilk.Class_train</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/classify.html">classify</a></span>(<span class="no">train_expression</span>[<span class="no">inBoth</span>,],<span class="no">centroids</span>[<span class="no">inBoth</span>,])
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">Wilk.Class_train</span>)
<span class="co">#&gt; Wilk.Class_train</span>
<span class="co">#&gt;   1   2   3 </span>
<span class="co">#&gt; 219 115 136</span>

<span class="no">Wilk.Class_test</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/classify.html">classify</a></span>(<span class="no">test_expression</span>[<span class="no">inBoth</span>,],<span class="no">centroids</span>[<span class="no">inBoth</span>,])
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">Wilk.Class_test</span>)
<span class="co">#&gt; Wilk.Class_test</span>
<span class="co">#&gt;   1   2   3 </span>
<span class="co">#&gt; 156 121 162</span></pre></body></html></div>
<p>Once the patients are classified according to their closest centroids, we can now evaluate the survival curves for the different types in each of the datasets</p>
<div id="survival-of-tcga-patients" class="section level3">
<h3 class="hasAnchor">
<a href="#survival-of-tcga-patients" class="anchor"></a>Survival of TCGA patients</h3>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">survival</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">survminer</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)


<span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(train_clinic$time,train_clinic$status)~ Wilk.Class_train"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"TCGA Lung adenocarcinoma \n Wilkerson subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1825</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="../inst/extdata/images/TCGA_wilk.jpg" width="100%"></p>
</div>
<div id="survival-of-gse68465-patients" class="section level3">
<h3 class="hasAnchor">
<a href="#survival-of-gse68465-patients" class="anchor"></a>Survival of GSE68465 patients</h3>
<div class="sourceCode"><html><body><pre class="r">

<span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$time,test_clinic$status)~ Wilk.Class_test"</span>)
<span class="no">tumortotal2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal2diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal2pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal2diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal2diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal2</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"GSE68465 Lung adenocarcinoma \n Wilkerson subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1825</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="../inst/extdata/images/GSE_wilk.jpg" width="100%"></p>
</div>
</div>
<div id="find-lung-adenocarcinoma-gene-signatures-with-galgor" class="section level2">
<h2 class="hasAnchor">
<a href="#find-lung-adenocarcinoma-gene-signatures-with-galgor" class="anchor"></a>Find lung adenocarcinoma gene signatures with GalgoR</h2>
<p>Now we run Galgo to find cohesive and clinically meaningful signatures for LUAD using TCGA data as training set and GSE68465 data as test set</p>
<div id="set-configuration-parameters" class="section level3">
<h3 class="hasAnchor">
<a href="#set-configuration-parameters" class="anchor"></a>Set configuration parameters</h3>
<div class="sourceCode"><html><body><pre class="r"><span class="no">population</span> <span class="kw">&lt;-</span> <span class="fl">150</span>
<span class="no">generations</span> <span class="kw">&lt;-</span><span class="fl">50</span>
<span class="no">nCV</span> <span class="kw">&lt;-</span> <span class="fl">5</span>
<span class="no">distancetype</span> <span class="kw">&lt;-</span> <span class="st">"pearson"</span>
<span class="no">TournamentSize</span> <span class="kw">&lt;-</span> <span class="fl">2</span>
<span class="no">period</span> <span class="kw">&lt;-</span> <span class="fl">1825</span>
<span class="no">usegpu</span> <span class="kw">&lt;-</span> <span class="fl">TRUE</span></pre></body></html></div>
<p>Run Galgo on the training set</p>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">output</span><span class="kw">=</span> <span class="kw pkg">galgoR</span><span class="kw ns">::</span><span class="fu"><a href="../reference/galgo.html">galgo</a></span>(<span class="kw">generations</span> <span class="kw">=</span> <span class="no">generations</span>, <span class="kw">population</span> <span class="kw">=</span> <span class="no">population</span>,<span class="kw">prob_matrix</span> <span class="kw">=</span> <span class="no">train_expression</span>, <span class="kw">OS</span><span class="kw">=</span><span class="no">train_surv</span>,<span class="kw">usegpu</span><span class="kw">=</span><span class="no">usegpu</span>,<span class="kw">nCV</span><span class="kw">=</span> <span class="no">nCV</span>, <span class="kw">distancetype</span><span class="kw">=</span><span class="no">distancetype</span>, <span class="kw">TournamentSize</span><span class="kw">=</span><span class="no">TournamentSize</span>, <span class="kw">period</span><span class="kw">=</span><span class="no">period</span>)</pre></body></html></div>
<pre><code>#&gt; [1] "galgo.Obj"
#&gt; attr(,"package")
#&gt; [1] "galgoR"</code></pre>
</div>
</div>
<div id="analyzing-galgo-results" class="section level2">
<h2 class="hasAnchor">
<a href="#analyzing-galgo-results" class="anchor"></a>Analyzing Galgo results</h2>
<div id="pareto-front" class="section level3">
<h3 class="hasAnchor">
<a href="#pareto-front" class="anchor"></a>Pareto front</h3>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_pareto.html">plot_pareto</a></span>(<span class="no">output</span>)</pre></body></html></div>
<p><img src="../inst/extdata/images/pareto2.jpg" width="100%"></p>
</div>
<div id="summary-of-the-results" class="section level3">
<h3 class="hasAnchor">
<a href="#summary-of-the-results" class="anchor"></a>Summary of the results</h3>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">output_df</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/toDataFrame.html">toDataFrame</a></span>(<span class="no">output</span>)

<span class="no">NonDom_solutions</span><span class="kw">&lt;-</span> <span class="no">output_df</span>[<span class="no">output_df</span>$<span class="no">Rank</span><span class="kw">==</span><span class="fl">1</span>,]

<span class="co"># N of non-dominated solutions </span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">NonDom_solutions</span>)
<span class="co">#&gt; [1] 17</span>

<span class="co"># N of partitions found</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">k</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt;  2  3  4  8 </span>
<span class="co">#&gt; 12  2  1  2</span>

<span class="co">#Average N of genes per signature</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">Genes</span>,<span class="no">length</span>)))
<span class="co">#&gt; [1] 82.17647</span>

<span class="co">#SC range</span>
<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">SC.Fit</span>)
<span class="co">#&gt; [1] 0.0799691 0.2745406</span>

<span class="co"># Survival fitnesss range</span>
<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="no">NonDom_solutions</span>$<span class="no">Surv.Fit</span>)
<span class="co">#&gt; [1] 224.8008 533.7608</span></pre></body></html></div>
</div>
<div id="select-best-performing-solutions" class="section level3">
<h3 class="hasAnchor">
<a href="#select-best-performing-solutions" class="anchor"></a>Select best performing solutions</h3>
<p>Now we select the best performing solutions for each number of partitions (k) according to C.Index</p>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">RESULT</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/non_dominated_summary.html">non_dominated_summary</a></span>(<span class="kw">output</span><span class="kw">=</span><span class="no">output</span>,<span class="kw">OS</span><span class="kw">=</span><span class="no">train_surv</span>, <span class="kw">prob_matrix</span><span class="kw">=</span> <span class="no">train_expression</span>, <span class="kw">distancetype</span> <span class="kw">=</span><span class="no">distancetype</span>, <span class="kw">usegpu</span><span class="kw">=</span> <span class="no">usegpu</span> )


<span class="no">best_sol</span><span class="kw">=</span><span class="kw">NULL</span>
  <span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="no">RESULT</span>$<span class="no">k</span>)){
  <span class="no">best_sol</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">best_sol</span>,<span class="no">RESULT</span>[<span class="no">RESULT</span>$<span class="no">k</span><span class="kw">==</span><span class="no">i</span>,<span class="st">"solution"</span>][<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="no">RESULT</span>[<span class="no">RESULT</span>$<span class="no">k</span><span class="kw">==</span><span class="no">i</span>,<span class="st">"C.Index"</span>])])
  }

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">best_sol</span>)
<span class="co"># [1] "Solutions.1" "Solutions.7" "Solutions.3" "Solutions.9"   </span></pre></body></html></div>
</div>
<div id="create-prototypic-centroids" class="section level3">
<h3 class="hasAnchor">
<a href="#create-prototypic-centroids" class="anchor"></a>Create prototypic centroids</h3>
<p>Now we create the prototypic centroids of the selected solutions</p>
<div class="sourceCode"><html><body><pre class="r">  <span class="no">CentroidsList</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create_centroids.html">create_centroids</a></span>(<span class="no">output</span>, <span class="kw">solution.names</span> <span class="kw">=</span> <span class="no">best_sol</span>, <span class="kw">train.set</span> <span class="kw">=</span> <span class="no">train_expression</span>)</pre></body></html></div>
</div>
</div>
<div id="test-galgo-signatures-in-a-test-set" class="section level2">
<h2 class="hasAnchor">
<a href="#test-galgo-signatures-in-a-test-set" class="anchor"></a>Test Galgo signatures in a test set</h2>
<p>We will test the Galgo signatures found with the TCGA training set in an independent test set</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">testSet</span><span class="kw">=</span><span class="no">rna_luad</span>$<span class="no">GSE68465</span>
<span class="no">prob_matrix_test</span><span class="kw">=</span> <span class="no">testSet</span>$<span class="no">expression_matrix</span>
<span class="no">clinical_test</span><span class="kw">=</span><span class="no">testSet</span>$<span class="no">pheno_data</span>
<span class="no">OS_test</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span>(<span class="kw">time</span><span class="kw">=</span><span class="no">clinical_test</span>$<span class="no">time</span>,<span class="kw">event</span><span class="kw">=</span><span class="no">clinical_test</span>$<span class="no">status</span>)</pre></body></html></div>
<div id="classify-train-and-test-set-into-galgor-subtypes" class="section level3">
<h3 class="hasAnchor">
<a href="#classify-train-and-test-set-into-galgor-subtypes" class="anchor"></a>Classify train and test set into galgoR subtypes</h3>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">train_classes</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/classify_multiple.html">classify_multiple</a></span>(<span class="kw">prob_matrix</span><span class="kw">=</span><span class="no">train_expression</span>,<span class="kw">centroid.list</span><span class="kw">=</span> <span class="no">CentroidsList</span>, <span class="kw">distancetype</span> <span class="kw">=</span> <span class="no">distancetype</span>)


<span class="no">test_classes</span><span class="kw">&lt;-</span> <span class="fu"><a href="../reference/classify_multiple.html">classify_multiple</a></span>(<span class="kw">prob_matrix</span><span class="kw">=</span><span class="no">test_expression</span>,<span class="kw">centroid.list</span><span class="kw">=</span> <span class="no">CentroidsList</span>, <span class="kw">distancetype</span> <span class="kw">=</span> <span class="no">distancetype</span>)</pre></body></html></div>
</div>
<div id="calculate-train-and-test-set-c-index" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-train-and-test-set-c-index" class="anchor"></a>Calculate train and test set C.Index</h3>
<p>To calculate the train and test C.Index, the risk coefficients are calculated for each subclass in the training set and are then used to predict the risk of the different groups in the test set. This is particularly important for signatures with high number of partitions, were the survival differences of different groups might overlap and change their relative order, which is of great importance in the C.Index calculation.</p>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">Prediction.models</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="no">best_sol</span>){

  <span class="no">OS</span><span class="kw">&lt;-</span> <span class="no">train_surv</span>
  <span class="no">predicted_class</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">train_classes</span>[,<span class="no">i</span>])
  <span class="no">predicted_classdf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">predicted_class</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">predicted_classdf</span>)<span class="kw">&lt;-</span> <span class="no">i</span>

  <span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"OS~ "</span>,<span class="no">i</span>))
  <span class="no">coxsimple</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/coxph.html">coxph</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">predicted_classdf</span>)


<span class="no">Prediction.models</span><span class="kw">[[</span><span class="no">i</span>]]<span class="kw">&lt;-</span> <span class="no">coxsimple</span>

}</pre></body></html></div>
</div>
<div id="calculate-c-index-for-training-and-test-set-using-the-prediction-models-of-the-training-set" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-c-index-for-training-and-test-set-using-the-prediction-models-of-the-training-set" class="anchor"></a>Calculate C.Index for training and test set using the prediction models of the training set</h3>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">C.indexes</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">train_CI</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">NA</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">best_sol</span>)),<span class="kw">test_CI</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">NA</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">best_sol</span>)))
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">C.indexes</span>)<span class="kw">&lt;-</span> <span class="no">best_sol</span>

<span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="no">best_sol</span>){

<span class="no">predicted_class_train</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">train_classes</span>[,<span class="no">i</span>])
    <span class="no">predicted_class_train_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">predicted_class_train</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">predicted_class_train_df</span>)<span class="kw">&lt;-</span> <span class="no">i</span>

<span class="no">CI_train</span><span class="kw">&lt;-</span> <span class="fu">concordance.index</span>(<span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span>(<span class="no">Prediction.models</span><span class="kw">[[</span><span class="no">i</span>]],<span class="no">predicted_class_train_df</span>),<span class="kw">surv.time</span><span class="kw">=</span><span class="no">train_surv</span>[,<span class="fl">1</span>],<span class="kw">surv.event</span><span class="kw">=</span><span class="no">train_surv</span>[,<span class="fl">2</span>],<span class="kw">outx</span><span class="kw">=</span><span class="fl">FALSE</span>)$<span class="no">c.index</span>

<span class="no">C.indexes</span>[<span class="no">i</span>,<span class="st">"train_CI"</span>]<span class="kw">&lt;-</span> <span class="no">CI_train</span>

<span class="no">predicted_class_test</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">test_classes</span>[,<span class="no">i</span>])
    <span class="no">predicted_class_test_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">predicted_class_test</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">predicted_class_test_df</span>)<span class="kw">&lt;-</span> <span class="no">i</span>


<span class="no">CI_test</span><span class="kw">&lt;-</span> <span class="fu">concordance.index</span>(<span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span>(<span class="no">Prediction.models</span><span class="kw">[[</span><span class="no">i</span>]],<span class="no">predicted_class_test_df</span>),<span class="kw">surv.time</span><span class="kw">=</span><span class="no">test_surv</span>[,<span class="fl">1</span>],<span class="kw">surv.event</span><span class="kw">=</span><span class="no">test_surv</span>[,<span class="fl">2</span>],<span class="kw">outx</span><span class="kw">=</span><span class="fl">FALSE</span>)$<span class="no">c.index</span>

<span class="no">C.indexes</span>[<span class="no">i</span>,<span class="st">"test_CI"</span>]<span class="kw">&lt;-</span> <span class="no">CI_test</span>

}

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">C.indexes</span>)

<span class="co">#             train_CI   test_CI</span>
<span class="co">#Solutions.1 0.6079561 0.5685536</span>
<span class="co">#Solutions.7 0.6034458 0.6025023</span>
<span class="co">#Solutions.3 0.6147336 0.6060875</span>
<span class="co">#Solutions.9 0.6024516 0.5916318</span>


<span class="no">best_signature</span><span class="kw">&lt;-</span> <span class="no">best_sol</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="no">C.indexes</span>$<span class="no">test_CI</span>)]

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">best_signature</span>)
<span class="co"># "Solutions.3"</span></pre></body></html></div>
</div>
<div id="evaluate-prediction-survival-of-galgo-signatures" class="section level3">
<h3 class="hasAnchor">
<a href="#evaluate-prediction-survival-of-galgo-signatures" class="anchor"></a>Evaluate prediction survival of Galgo signatures</h3>
<p>We test best galgo signature with training and test sets</p>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">train_class</span> <span class="kw">&lt;-</span> <span class="no">train_classes</span>[,<span class="no">best_signature</span>]

<span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(train_clinic$time,train_clinic$status)~ train_class"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">train_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"TCGA Lung adenocarcinoma \n Galgo subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1825</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="../inst/extdata/images/TCGA_galgo.jpg" width="100%"></p>
<div class="sourceCode"><html><body><pre class="r">
<span class="no">test_class</span> <span class="kw">&lt;-</span> <span class="no">test_classes</span>[,<span class="no">best_signature</span>]

<span class="no">surv_formula</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$time,test_clinic$status)~ test_class"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

  <span class="no">p</span><span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">tumortotal1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"GSE68465 Lung adenocarcinoma \n Galgo subtypes survival"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1825</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="../inst/extdata/images/GSE_galgo.jpg" width="100%"></p>
</div>
</div>
<div id="comparison-of-galgo-vs-wilkerson-classifiers" class="section level2">
<h2 class="hasAnchor">
<a href="#comparison-of-galgo-vs-wilkerson-classifiers" class="anchor"></a>Comparison of Galgo vs Wilkerson classifiers</h2>
<p>Compare Wilkerson classification vs Galgo classification in the GSE68465 (test) dataset</p>
<div class="sourceCode"><html><body><pre class="r">

<span class="no">surv_formula1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$time,test_clinic$status)~ test_class"</span>)
<span class="no">tumortotal1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula1</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal1diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula1</span>)
<span class="no">tumortotal1pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="no">surv_formula2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span>(<span class="st">"Surv(test_clinic$time,test_clinic$status)~ Wilk.Class_test"</span>)
<span class="no">tumortotal2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survminer/man/surv_fit.html">surv_fit</a></span>(<span class="no">surv_formula2</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>)
<span class="no">tumortotal2diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survdiff.html">survdiff</a></span>(<span class="no">surv_formula2</span>)
<span class="no">tumortotal2pval</span><span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html">pchisq</a></span>(<span class="no">tumortotal1diff</span>$<span class="no">chisq</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">tumortotal2diff</span>$<span class="no">n</span>) - <span class="fl">1</span>, <span class="kw">lower.tail</span> <span class="kw">=</span> <span class="fl">FALSE</span>)




  <span class="no">SURV</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">GALGO</span><span class="kw">=</span><span class="no">tumortotal1</span>,<span class="kw">Wilk</span><span class="kw">=</span><span class="no">tumortotal2</span> )
  <span class="no">COLS</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>:<span class="fl">8</span>,<span class="fl">10</span>)
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">cex</span><span class="kw">=</span><span class="fl">1.35</span>, <span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">3.8</span>, <span class="fl">3.8</span>, <span class="fl">2.5</span>, <span class="fl">2.5</span>) + <span class="fl">0.1</span>)
  <span class="no">p</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span>(<span class="no">SURV</span>,<span class="kw">combine</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">data</span><span class="kw">=</span><span class="no">test_clinic</span>,<span class="kw">risk.table</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">pval</span><span class="kw">=</span><span class="fl">TRUE</span>,<span class="kw">palette</span><span class="kw">=</span><span class="st">"dark2"</span>, <span class="kw">title</span><span class="kw">=</span><span class="st">"Galgo vs. Wilkerson subtypes \n Lung survival comparison"</span>, <span class="kw">surv.scale</span><span class="kw">=</span><span class="st">"percent"</span>, <span class="kw">conf.int</span><span class="kw">=</span><span class="fl">FALSE</span>, <span class="kw">xlab</span><span class="kw">=</span><span class="st">"time (days)"</span>, <span class="kw">ylab</span><span class="kw">=</span><span class="st">"survival(%)"</span>, <span class="kw">xlim</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="no">period</span>),<span class="kw">break.time.by</span> <span class="kw">=</span> <span class="fl">365</span>, <span class="kw">ggtheme</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span>(), <span class="kw">risk.table.y.text.col</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">risk.table.y.text</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">censor</span><span class="kw">=</span><span class="fl">FALSE</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<p><img src="../inst/extdata/images/GSE_galgovswilk.jpg" width="100%"> # Session info</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 3.6.1 (2019-07-05)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 16.04.5 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/libblas/libblas.so.3.6.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/lapack/liblapack.so.3.6.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] galgoR_0.2.0      survival_2.44-1.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.3       rstudioapi_0.11  knitr_1.25       magrittr_1.5    </span>
<span class="co">#&gt;  [5] splines_3.6.1    MASS_7.3-51.4    lattice_0.20-38  R6_2.4.1        </span>
<span class="co">#&gt;  [9] rlang_0.4.6      stringr_1.4.0    tools_3.6.1      grid_3.6.1      </span>
<span class="co">#&gt; [13] xfun_0.10        htmltools_0.4.0  yaml_2.2.0       assertthat_0.2.1</span>
<span class="co">#&gt; [17] digest_0.6.25    rprojroot_1.3-2  pkgdown_1.5.1    crayon_1.3.4    </span>
<span class="co">#&gt; [21] Matrix_1.2-17    fs_1.3.1         memoise_1.1.0    evaluate_0.14   </span>
<span class="co">#&gt; [25] rmarkdown_2.1    stringi_1.4.3    compiler_3.6.1   desc_1.2.0      </span>
<span class="co">#&gt; [29] backports_1.1.5</span></pre></body></html></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Carlos Catania.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
